VoiceController.php

```diff
@@ -1,48 +1,30 @@
-public function voice(Request $request){
-    $request->validate([
-        'question_id'=>'required|int|exists:questions,id',
-        'value'=>'required|boolean',
-    ]);
 
+
+public function voice(StoreQuestionRequest $request){
     $question=Question::find($request->post('question_id'));
-    if (!$question)
-        return response()->json([
-            'status'=>404,
-            'message'=>'not found question ..'
-        ]);
-    if ($question->user_id==auth()->id())
-        return response()->json([
-            'status' => 500,
-            'message' => 'The user is not allowed to vote to your question'
+    if (!$question->user_id==auth()->id()){
+        if($question->voice()->user_id == auth()->id()){
+             $question->voice()->value = $request->post('value');
+             if($voice->isDirty()){
```

Comment:
@javiershaka I think you introduce `$voice` variable here without initializing it.

Reply:
I forgot to change that variable when I made the refactor, I will make the corrections so that they do not remain as a bad example.